Homework2/Task1MatrixClass/Matrix.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace Task1MatrixClass
+{
+    public class Matrix
+    {
+        private int _size;
+        private int[] _diagonalArray;
+
+        public int Size { get => _size; set => _size = value; }
+        public int[] DiagonalArray { get => _diagonalArray; set => _diagonalArray = value; }
+
+        public Matrix(int matrixSize, params int[] diagonalNumbers)
+        {
+            Size = diagonalNumbers.Length;
```

Comment:
do not put matrixSize. Just make ctor as 

public Matrix(params int[] diagonalNumbers)
{
if (diagonalNumbers == null)
            {
                Size = 0;
            }
   Size = diagonalNumbers.Length;
  DiagonalArray = diagonalNumbers;

}